public/jira/derived/work-status/work-status.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * This module is repsonsible for determining the correct workType ("design", "dev", "qa", "uat")
+ * and statusType ("qa", "uat", "todo", "done", "blocked") for an issue.
+ */
+export const workType = ["design", "dev", "qa", "uat"] as const;
+export const workTypes = workType;
+
+export const workflowHappyPath = ["todo", "design", "dev", "qa", "uat", "done"];
+export const workflowUnhappyStatuses = ["blocked"];
+
+const inQAStatus = { qa: true, "in qa": true, "qa complete": true };
+const inPartnerReviewStatus = { "partner review": true, uat: true };
+const inDoneStatus = { done: true, cancelled: true };
+const blockedStatus = { blocked: true, delayed: true };
+
+export const inPartnerReviewStatuses = Object.keys(inPartnerReviewStatus);
+export const inIdeaStatus = { idea: true, "to do": true, open: true };
+export const inIdeaStatuses = Object.keys(inIdeaStatus);
+
+type WorkType = (typeof workType)[number];
+
+type Status =
+  | keyof typeof inQAStatus
+  | keyof typeof inPartnerReviewStatus
+  | keyof typeof inIdeaStatus
+  | keyof typeof inDoneStatus
+  | keyof typeof blockedStatus
+  | (string & {});
+
+type StatusCategory = "qa" | "uat" | "todo" | "done" | "blocked";
+
+export const statusCategoryMap = (function () {
+  const items = [
+    ["qa", inQAStatus],
+    ["uat", inPartnerReviewStatus],
+    ["todo", inIdeaStatus],
+    ["done", inDoneStatus],
+    ["blocked", blockedStatus],
+  ] as const;
+
+  const statusCategoryMap = {} as Record<Status, StatusCategory>;
+
+  for (let [category, statusMap] of items) {
+    for (let prop in statusMap) {
+      const status = prop;
+      statusCategoryMap[status] = category;
+    }
+  }
+
+  return statusCategoryMap;
+})();
+
+export function getStatusTypeDefault(issue: { status?: string }): StatusCategory | "dev" {
+  const statusCategory = statusCategoryMap[toLowerCase(issue?.status || "")];
+  if (statusCategory) {
+    return statusCategory;
+  } else {
+    return "dev";
```

Comment:
if `dev` is not a Status why are we returning it as the default value for a StatusType.
Should it be added to the StatusCategory?

Reply:
Hmmm yeah probably

Reply:
Oh wait, yeah sorry i remember now why I did it this way. Its not used in the `statusCategoryMap` I could add it to the union and `Omit` it, or I could leave it this way. Whichever you prefer I'm fine with. I felt this way read easier

Reply:
Yes my confusion was about why `dev` is  being returned as a possible value of `getStatusTypeDefault` although its not part of the `StatusCategory` types (is it one of the issues statuses in Jira?). 

But seems that this is the case.  So I am fine with the current implementation where the return type is `StatusCategory | "dev"`